File: standalone/module2Standalone/src/main/java/io/github/prototypez/module2/standalone/Module2StandaloneApplication.java
package io.github.prototypez.module2.standalone;

import android.util.Log;

import io.github.prototypez.appjoint.module2.Module2Application;

public class Module2StandaloneApplication extends Module2Application {

    @Override
    public void onCreate() {
        // module1 init inside super.onCreate()
        super.onCreate();
        // initialization only used for running module1 standalone
        Log.i("module2Standalone", "module2Standalone init is called");
    }
}

File: standalone/module2Standalone/src/main/java/io/github/prototypez/appjoint/module2/standalone/Module2StandaloneApplication.kt
package io.github.prototypez.appjoint.module2.standalone

import android.util.Log

import io.github.prototypez.appjoint.module2.Module2Application
import io.github.prototypez.appjoint.module2.Services
import io.github.prototypez.appjoint.module2.standalone.mock.AppServiceMock
import io.github.prototypez.appjoint.module2.standalone.mock.Module1ServiceMock

class Module2StandaloneApplication : Module2Application() {

  override fun onCreate() {
    // module2 init inside super.onCreate()
    super.onCreate()
    // initialization only used for running module2 standalone
    Log.i("module2Standalone", "module2Standalone init is called")
    Services.sAppService = AppServiceMock()
    Services.sModule1Service = Module1ServiceMock()
  }
}
